### DISCUSSION TITLE
Template Builder

### DISCUSSION THREAD

**Discussion by antoninlanglade** [unlikelystudio/bases#14] [Show and tell]
### Hello @unlikelystudio/front-end-team & @unlikelystudio/back-end-team 🚀,

Suite à une proposition de @taiwabisabi dans le doc d'externalisation, j'ai commencé à construire un **_package de templating via CLI_** pour les actions très récurrentes et désagréables du quotidien au sein d'un projet.

J'ai pour le moment publié ce package `@unlikelystudio/bases-plop` qui permet de générer trois types d'entitées : 

- _Next Page_ (single or multiple, with or without slice manager)
- _Component_
- _Slice_

![image](https://user-images.githubusercontent.com/1774288/113872109-ddbca500-97b3-11eb-841c-114e317e5ac9.png)
![image](https://user-images.githubusercontent.com/1774288/113872200-f7f68300-97b3-11eb-97f6-9435309fbdbc.png)
![image](https://user-images.githubusercontent.com/1774288/113872321-12c8f780-97b4-11eb-8e56-ddf7ed293af9.png)

J'ai découpé les différents types de générateurs dans le dossier `/templates/`

@baloran n'hésites pas si tu as des insights pour la partie back-end tu as peut être également de ton côté des tâches très redondantes à automatiser.

Je suis ouvert à toutes vos propositions.

👨🏻‍💻

**Comment by bastienrobert** [comment]
`with or without slice manager` pas sur que ce soit nécessaire de prendre en charge le slice manager et d'y faire des imports automatisés. On pourrait envisager un retour dans la console pour donner le path du fichier créer plutôt ?

**Reply by antoninlanglade** [reply]
Pour moi ça a un sens car ça te met automatiquer le slice manager avec son fichier d'import personnalisé.
Tu peux préciser pour le path ? Je ne comprend pas où tu veux en venir.

**Reply by bastienrobert** [reply]
C'est un peu magique mais ça se tente. 
Imaginons que les slices ou le SliceManager ne soient pas rangées à la racine du dossier composant mais dans un doud-dossier, est-ce qu'on peut configurer ça quelque part ?

**Reply by antoninlanglade** [reply]
On pourrait imaginer comme pour le reste effectivement de donner la possibilité de le configurer via file

**Reply by bastienrobert** [reply]
Ça pourrait être pas mal d'avoir un plop-template.config.js dans lequel tu peux spécifier certains chemins de dossier je pense, à investiguer pour une v2 maybe ?

**Reply by antoninlanglade** [reply]
complètement d'accord, si l'utilitaire est vraiment utilisé il faudra le rendre plus configurable

**Comment by bastienrobert** [comment]
Au niveau du path dans lequel les fichiers sont crées, est-ce qu'on pourrait avoir un fichier de configuration pour les préciser ? Typiquement préciser où est-ce que les slices sont générées.

**Comment by bastienrobert** [comment]
On pourrait aussi envisager de sortir ce paquet de `bases` et le garder dans un `@unlikelystudio/plop-templates` ?
